### PR TITLE
fixing issue 6989

### PR DESCRIPTION
**Pull request by shettyvarun268**
Fixes: #6989 

Deployments can falsely exit with success code 0 even when internal tasks fail. Right before shutting down, the CLI tries to send background analytics. In secure or restrictive CI/CD environments, this analytics ping hangs and times out. That timeout crashes the CLI wrapper's error-handler prematurely, masking the original deployment failure before it can set the non-zero exit code.

Solution: We wrapped the background analytics timeouts in protective try...catch blocks. Now, if the telemetry tracker hangs or fails to reach the portals, the CLI simply ignores the analytics failure and continues with its shutdown routine. This ensures the CLI always returns the authentic deployment failure code to the host system.

Did not add new unit tests for this specific logic because the wrapper lives at the top-level process exit boundary. Writing automated tests here requires faking CLI process shutdowns and restricted networks, which creates brittle mocks. Verified the logic safety using an isolated sandbox playground and confirmed all existing test trees continue to pass.

### CODE REVIEW

File: src/command.ts [hunk 1]
```diff
@@ -236,28 +236,28 @@ export class Command {
...
+              tracks.push(
                 trackEmulator("command_success", {
                   command_name: this.name,
                   duration,
                 }),
-              ]),
-            );
+              );
+            }
+            await withTimeout(5000, Promise.all(tracks));
```
Comment (joehan) [inline src/command.ts]: Nit: 5s is insanely long to wait for these calls, probably safe to lower this to 1s - same for the other one.

**Review by joehan — APPROVED** [review]
LGTM, but please add a CHANGELOG entry saying something like:
```
- Fixed an issue where functions deployments would silently fail (#6989)
```
